ECS: Allow to Add ECS to Kitchensink-Forms

kergomard opened the pull request:
Hi @pascalseeland 
This PR adds an Option to add ECS-Settings as a KS-Input-Section to Settings-Forms. I did this, to enable the move of the general settings of the Test to KS-Forms.

The changes in `ECSSettingsGUI` are just a fix for an issue with types of POST-Values that came to be a problem because of the use of strict types.
The naming in `ilECSObjectSettings` is a little funky for my taste, I tried to find a workable naming, but let me know, if you would like it otherwise.

I tested the functionality with some spoofed data and everything works as far as I can see, but sadly, I don't have an ECS-Server at my disposal.

Thank you very much and best,
@kergomard

kergomard committed:
ECS: Allow to Add ECS to Kitchensink-Forms

kergomard committed:
ECS: Fix Type Error and Variable

kergomard commented:
@pascalseeland : Any chance of a feedback on this?

Thank you very much and best!

kergomard commented:
Hi @pascalseeland 

I would be very grateful if you could have a look at this.

Thanks and best,
@kergomard

pascalseeland commented on Services/WebServices/ECS/classes/class.ilECSObjectSettings.php:
```diff
@@ -119,31 +125,11 @@ protected function isActive(): bool
-            $this->logger->debug('Object type is not active. => no settings.');
-            return false;
-        }
+        $export_manager = ilECSExportManager::getInstance();
+        $exportable_participants = (new ilECSParticipantSettingsRepository())->getExportableParticipants($a_type);
```
This renames the used variables from exportManager to export_manager, which causes issues in the rest of this method. Same issue exists with exportableParticipants.

pascalseeland commented on Services/WebServices/ECS/classes/class.ilECSObjectSettings.php:
```diff
@@ -119,31 +125,11 @@ protected function isActive(): bool
+        $export_manager = ilECSExportManager::getInstance();
+        $exportable_participants = (new ilECSParticipantSettingsRepository())->getExportableParticipants($a_type);
 
-        $obj_id = $this->content_obj->getId();
```
This variable is used in the rest of this method. If it is removed here, the rest of the usages need to be changed as well.

pascalseeland reviewed (changes requested):
Please fix the missing renaming of variables in lines 145, 160 and 180 as it breaks the code.

kergomard committed:
ECS: Fix Naming and Missing Variable

kergomard commented:
Hi @pascalseeland 
Thank you very much for your review! you are right and I fixed the issues. I also renamed a function I introduced from `ecsIsActivatableForObject` to `ecsCanBeActivatedforObject`.

Thanks and best,
Stephan
